synapse/lib/layer.py

```diff
@@ -2153,73 +2211,52 @@ async def iterLayerNodeEdits(self):
                         edit = (EDIT_EDGE_ADD, (verb, n2iden), ())
                         nodeedits[2].append(edit)
 
-                    yield nodeedits
+                    yield tuple(nodeedits)
+
+                nodeedits = [buid, None, []]
 
             if flag == 0:
                 form, valu, stortype = s_msgpack.un(lval)
-
-                nodeedits = (buid, form, [])
-
-                edit = (EDIT_NODE_ADD, (valu, stortype), ())
-                nodeedits[2].append(edit)
+                nodeedits[1] = form
+                nodeedits[2].append((EDIT_NODE_ADD, (valu, stortype), ()))
                 continue
 
             if flag == 1:
-                if not nodeedits[0] == buid:
-                    nodeedits = (buid, None, [])
-
                 name = lkey[33:].decode()
                 valu, stortype = s_msgpack.un(lval)
-
-                edit = (EDIT_PROP_SET, (name, valu, None, stortype), ())
-                nodeedits[2].append(edit)
+                nodeedits[2].append((EDIT_PROP_SET, (name, valu, None, stortype), ()))
                 continue
 
             if flag == 2:
-                if not nodeedits[0] == buid:
-                    nodeedits = (buid, None, [])
-
                 name = lkey[33:].decode()
                 tagv = s_msgpack.un(lval)
-
-                edit = (EDIT_TAG_SET, (name, tagv, None), ())
-                nodeedits[2].append(edit)
+                nodeedits[2].append((EDIT_TAG_SET, (name, tagv, None), ()))
                 continue
 
             if flag == 3:
-                if not nodeedits[0] == buid:
-                    nodeedits = (buid, None, [])
-
                 tag, prop = lkey[33:].decode().split(':')
                 valu, stortype = s_msgpack.un(lval)
-
                 edit = (EDIT_TAGPROP_SET, (tag, prop, valu, None, stortype), ())
                 nodeedits[2].append(edit)
                 continue
 
             if flag == 9:
-                if not nodeedits[0] == buid:
-                    form = lval.decode()
-                    nodeedits = (buid, form, [])
-
-                elif nodeedits[1] is None:
-                    form = lval.decode()
-                    nodeedits = (nodeedits[0], form, nodeedits[2])
-
+                nodeedits[1] = lval.decode()
```

Comment:
???

Reply:
we're more efficiently constructing the `nodeedits` and this is the line that updates the form. see the removed lines above.